src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -131,6 +135,7 @@ public NeuralQueryBuilder(StreamInput in) throws IOException {
         if (isClusterOnOrAfterMinReqVersion(METHOD_PARAMS_FIELD.getPreferredName())) {
             this.methodParameters = MethodParametersParser.streamInput(in, MinClusterVersionUtil::isClusterOnOrAfterMinReqVersion);
         }
+        this.rescoreContext = RescoreParser.streamInput(in);
```

Comment:
don't we need a version check here for rescore context?

Reply:
https://github.com/opensearch-project/neural-search/pull/885#discussion_r1744646335

Reply:
Make sense. But this is going to create more dependency of k-NN on Neural. After this we should start adding right annotations on K-NN code on what should be used outside of k-NN and what should not be.